### PR TITLE
Add recipe for `semimap`

### PR DESCRIPTION
**Pull request by jjerphan**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by jjerphan** [comment]
I added support for the test suite.

I think we need to discuss how we handle the lack of a version identifier.

**Comment by JohanMabille** [comment]
We usually avoid to copy header files. A better practice is to do it via `make install`, which also installs a cmake package file. This way, you can explicitely state the dependency in the CMakeLists.txt of the downstream package:

```cmake
find_package(semimap REQUIRED)
```

This requires a projectConfig.cmake file that is usually configured in the CMakeLists.txt in the install section. You can take inspiration from [this config file](https://github.com/xtensor-stack/xtl/blob/master/xtlConfig.cmake.in) and the [corresponding section](https://github.com/xtensor-stack/xtl/blob/master/CMakeLists.txt#L98) in the CMakeLists.txt.

This repo does not have such a config file; when packaging these repos for conda-forge, we usually contribute upsteam and then we have two options:
- Wait for the contribution to be accepted and for a new release including it. This is fine when the upstream maintainers are very responsive and can tag a new release quickly
- patch the repo in the recipes and release the package; when a new release is published, with the contribution, the patch can be removed.

In the case of `semimap`, since the main repo does not appear to be matained, I woud suggest to contribute upstream, but also to the fork in the QuantStack org (since we use it for having tags).

**Review by JohanMabille — CHANGES_REQUESTED** [review]
We usually avoid to direcly copy header files. A better practice is to run a `make install` command, that will copy the header and a cmake package file. Thi way, we can explicitely state the dependency in the downstream CMakeLists.txt:

```cmake
find_package(semimap REQUIRED)
```

This package file is usually the result of a config file used in the install process. Here is an example of such a [config file](https://github.com/xtensor-stack/xtl/blob/master/xtlConfig.cmake.in) and the [section of CMakeLists](https://github.com/xtensor-stack/xtl/blob/master/CMakeLists.txt#L98) using it.

In this case, the source repository does not contain this package file. When packaging such a repo for conda-forge, we usually contribute upstream, and then we have two options:
- wait for a new release with the contribution included
- patch the repo in the feedstock, release the package. The patches will be droped when a new release with the contribution is out.

Since the original `semimap` repo doe snot seem actively maintain, and since we have a fork in the QuantStack org for the release tags, I suggest to update the fork directly and not patch the repo in the recipes. (A PR should be opened to the original repo too).

File: recipes/semimap/meta.yaml
Comment (jjerphan) [inline recipes/semimap/meta.yaml:10]: To me, the last remaining piece of complexity to address is to have a proper version identifier for semimap.

https://github.com/QuantStack/semimap/commit/3e2175dbaf3f17004f91e0a0f26de39665607241 somewhat has introduced a v.1.0.0 version; after the first review of this recipe, the repository was restructured to use cmake.

Should we bump the version number, use a tag and create a release before this recipe gets approved? What do you think, @JohanMabille?

Comment (JohanMabille) [reply recipes/semimap/meta.yaml:10]: Yes, we should tag a release on the current commit (since the version is already set) and use it for this PR.

Comment (jjerphan) [reply recipes/semimap/meta.yaml:10]: What would be an appropriate version identifier? v1.1.0?

**Pull request merged**
